modules/osdk-monitoring-custom-metrics.adoc

```diff
@@ -0,0 +1,158 @@
+// Module included in the following assemblies:
+//
+// * operators/operator_sdk/osdk-monitoring-prometheus.adoc
+
+[id="osdk-monitoring-custom-metrics_{context}"]
+= Exposing custom metrics
+
+As an Operator author, you can publish custom metrics by using the global Prometheus registry from the `controller-runtime/pkg/metrics` library.
+
+.Prerequisites
+
+* Go-based Operator generated using the Operator SDK
+* Prometheus Operator (deployed by default on {product-title} clusters)
+
+.Procedure
+
+. In your Operator SDK project, uncomment the following line in the `config/default/kustomization.yaml` file:
++
+[source,yaml]
+----
+../prometheus
+----
+
+. Create a custom controller class to publish additional metrics from the Operator. The following example declares the `widgets` and `widgetFailures` collectors as global variables, and then registers them with the `init()` function in the controller's package:
++
+.`controllers/memcached_controller_test_metrics.go` file
+[%collapsible]
+====
+[source,go]
+----
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+
+var (
+    widgets = prometheus.NewCounter(
+        prometheus.CounterOpts{
+            Name: "widgets_total",
+            Help: "Number of widgets processed",
+        },
+    )
+    widgetFailures = prometheus.NewCounter(
+        prometheus.CounterOpts{
+            Name: "widget_failures_total",
+            Help: "Number of failed widgets",
+        },
+    )
+)
+
+func init() {
+    // Register custom metrics with the global prometheus registry
+    metrics.Registry.MustRegister(widgets, widgetFailures)
+}
+----
+====
+
+. Record to these collectors from any part of the reconcile loop in the `main` controller class, which determines the business logic for the metric:
++
+.`controllers/memcached_controller.go` file
+[%collapsible]
+====
+[source,go]
+----
+func (r *MemcachedReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	...
+	...
+	// Add metrics
+	widgets.Inc()
+	widgetFailures.Inc()
+
+	return ctrl.Result{}, nil
+}
+----
+====
+
+. Deploy the Operator:
++
+[source,terminal]
+----
+$ make deploy
```

Comment:
The default IMG is "controller:latest", should add the IMG build cmd
`
make docker-build docker-push IMG=<registry>/<user>/<image_name>:<tag>
make deploy IMG=<registry>/<user>/<image_name>:<tag>
`